### PR TITLE
feat: handle lowercase addresses

### LINKED ISSUE #172: Handle addresses not in checksum format
### Context

ETH addresses use a special pattern of casing as a [checksum](https://coincodex.com/article/2078/ethereum-address-checksum-explained/) against typos.

`xmtp-js` expects addresses in this format. If you want to initiate a conversation with an address that has been stored in lowercase or uppercase format, you can use ethers' [`getAddress`](https://docs.ethers.io/v5/api/utils/address/#utils-getAddress) utility function to convert the address to checksum format.

### Problem

When developers attempt to initiate conversations with lowercased/uppercased addresses, `xmtp-js` fails to do so and doesn't explain how to correct the issue. This can be confusing for developers who haven't previously had to convert addresses to checksum format.

### Goal

Make it possible to initiate conversations with lowercased/uppercased addresses without having prior knowledge about converting addresses to checksum format.

### Potential Solutions

1. Run `getAddress` on addresses as a convenience to developers (preferred)
2. Return an informative error message when a developer supplies an address that isn't in checksum format
3. Specify `xmtp-js`'s checksum format requirement in the README

### Notes

- This issue has been noted by developers while using `conversations.newConversation` and `Client.canMessage`.

### PR DESCRIPTION
**Pull request by dmccartney**
fixes #172

### CODE REVIEW

File: src/store/EncryptedStore.ts [hunk 1]
```diff
@@ -29,7 +29,8 @@ export default class EncryptedKeyStore implements KeyStore {
   private async getStorageAddress(name: string): Promise<string> {
     // I think we want to namespace the storage address by wallet
     // This will allow us to support switching between multiple wallets in the same browser
-    const walletAddress = await this.signer.getAddress()
+    let walletAddress = await this.signer.getAddress()
+    walletAddress = utils.getAddress(walletAddress)
```
[the remark this answers is not in the document]
Comment (dmccartney) [reply src/store/EncryptedStore.ts:33]: I know `ethers` implemented `Signers` are good about only returning checksummed addresses. But other implementations of `Signer` can mess it up (I got bit by this in another project).

File: src/Client.ts [hunk 2]
```diff
@@ -250,8 +250,8 @@ export default class Client {
   async getUserContact(
     peerAddress: string
   ): Promise<PublicKeyBundle | SignedPublicKeyBundle | undefined> {
+    peerAddress = utils.getAddress(peerAddress) // EIP55 normalize the address case.
```
[the remark this answers is not in the document]
Comment (dmccartney) [reply src/Client.ts:253]: 👍

File: src/utils/topic.ts [hunk 3]
```diff
@@ -1,11 +1,14 @@
...
 export const buildContentTopic = (name: string): string =>
   `/xmtp/0/${name}/proto`
 
 export const buildDirectMessageTopic = (
   sender: string,
   recipient: string
 ): string => {
-  const members = [sender, recipient]
+  // EIP55 normalize the address case.
+  const members = [utils.getAddress(sender), utils.getAddress(recipient)]
```
[the remark this answers is not in the document]
Comment (dmccartney) [reply src/utils/topic.ts:11]: I like the idea of normalizing / rejecting addresses as soon as possible. But I think it's still a good idea to call this pretty frequently throughout. It acts as a no-op assert for internal calls and will make it easier for us to detect where we fail to keep things tidy on the inside.

File: test/conversations/Conversation.test.ts [hunk 4]
```diff
@@ -248,7 +248,13 @@ describe('conversation', () => {
 
     it('throws when opening a conversation with an unknown address', () => {
       expect(alice.conversations.newConversation('0xfoo')).rejects.toThrow(
-        'Recipient 0xfoo is not on the XMTP network'
+        'invalid address'
+      )
+      const validButUnknown = '0x1111111111222222222233333333334444444444'
```
[the remark this answers is not in the document]
Comment (dmccartney) [reply test/conversations/Conversation.test.ts:253]: 👍

**Review by elisealix22 — APPROVED** [review]
This has been bugging us for a while so thank you so much! Looks good to me!

**Pull request merged**